File: app/src/test/java/com/dpis/module/MainActivityFilterStateSourceSmokeTest.java
package com.dpis.module;

import org.junit.Test;

import java.io.IOException;

import static org.junit.Assert.assertTrue;

public class MainActivityFilterStateSourceSmokeTest {
    @Test
    public void mainActivityLoadsAndSavesPersistedFilterState() throws IOException {
        String source = read("src/main/java/com/dpis/module/MainActivity.java");

        assertTrue(source.contains("private AppListFilterStateStore appListFilterStateStore;"));
        assertTrue(source.contains("appListFilterStateStore = new AppListFilterStateStore(this);"));
        assertTrue(source.contains("AppListFilterState initialFilterState = appListFilterStateStore.load();"));
        assertTrue(source.contains("initialFilterState = retainedState.filterState;"));
        assertTrue(source.contains("initialFilterState = new AppListFilterState("));
        assertTrue(source.contains("appListFilterStateStore.save(filterState);"));
        assertTrue(source.contains("dispatchMainUiAction(MainUiAction.filterChanged(filterState));"));
    }

    private static String read(String relativePath) throws IOException {
        return SourceSmokeTestPaths.read(relativePath);
    }
}

File: app/src/test/java/com/dpis/module/shell/MainActivityFilterStateSourceSmokeTest.kt
package com.dpis.module

import org.junit.Assert.assertTrue
import org.junit.Test

class MainActivityFilterStateSourceSmokeTest {
    @Test
    fun mainActivityLoadsAndSavesPersistedFilterState() {
        val startup = read(
            "src/main/java/com/dpis/module/ui/presentation/MainStartupSession.kt"
        )

        assertTrue(startup.contains("AppListFilterStateStore(activity)"))
        assertTrue(startup.contains("filterStore.load()"))
        assertTrue(startup.contains("filterState = retained.filterState"))
        assertTrue(startup.contains("filterState = AppListFilterState("))
        val filterSession = read(
            "src/main/java/com/dpis/module/applist/presentation/AppListFilterSession.kt"
        )
        val filterShell = read(
            "src/main/java/com/dpis/module/applist/presentation/AppListFilterShell.kt"
        )
        assertTrue(filterShell.contains("startupSession.filterStore?.save(filterState)"))
        assertTrue(filterSession.contains("MainUiAction.filterChanged(filterState)"))
    }

    private fun read(relativePath: String): String = SourceSmokeTestPaths.read(relativePath)
}
